src/commonMain/kotlin/fr/acinq/lightning/crypto/RouteBlinding.kt

```diff
@@ -0,0 +1,118 @@
+
+    /**
+     * @param publicKey            introduction node's public key (which cannot be blinded since the sender need to find a route to it).
+     * @param blindedPublicKey     blinded public key, which hides the real public key.
+     * @param blindingEphemeralKey blinding tweak that can be used by the receiving node to derive the private key that
+     *                             matches the blinded public key.
+     * @param encryptedPayload     encrypted payload that can be decrypted with the introduction node's private key and the
+     *                             blinding ephemeral key.
+     */
+    data class IntroductionNode(
+        val publicKey: PublicKey,
+        val blindedPublicKey: PublicKey,
+        val blindingEphemeralKey: PublicKey,
+        val encryptedPayload: ByteVector
+    )
+
+    /**
+     * @param blindedPublicKey blinded public key, which hides the real public key.
+     * @param encryptedPayload encrypted payload that can be decrypted with the receiving node's private key and the
+     *                         blinding ephemeral key.
+     */
+    data class BlindedNode(val blindedPublicKey: PublicKey, val encryptedPayload: ByteVector)
+
+    /**
+     * @param introductionNodeId the first node, not blinded so that the sender can locate it.
+     * @param blindingKey        blinding tweak that can be used by the introduction node to derive the private key that
+     *                           matches the blinded public key.
+     * @param blindedNodes       blinded nodes (including the introduction node).
+     */
+    data class BlindedRoute(
+        val introductionNodeId: PublicKey,
+        val blindingKey: PublicKey,
+        val blindedNodes: List<BlindedNode>
+    ) {
+        val introductionNode: IntroductionNode = IntroductionNode(
+            introductionNodeId,
+            blindedNodes.first().blindedPublicKey,
+            blindingKey,
+            blindedNodes.first().encryptedPayload
+        )
+        val subsequentNodes: List<BlindedNode> = blindedNodes.drop(1)
+        val blindedNodeIds: List<PublicKey> = blindedNodes.map { it.blindedPublicKey }
+        val encryptedPayloads: List<ByteVector> = blindedNodes.map { it.encryptedPayload }
+    }
+
+    /**
+     * Blind the provided route and encrypt intermediate nodes' payloads.
+     *
+     * @param sessionKey this node's session key.
+     * @param publicKeys public keys of each node on the route, starting from the introduction point.
+     * @param payloads   payloads that should be encrypted for each node on the route.
+     * @return a blinded route.
+     */
+    fun create(sessionKey: PrivateKey, publicKeys: List<PublicKey>, payloads: List<ByteVector>): BlindedRoute {
+        require(publicKeys.size == payloads.size) { "a payload must be provided for each node in the blinded path" }
+        var e = sessionKey
+        val (blindedHops, blindingKeys) = publicKeys.zip(payloads).map {
+            pair ->
```

Comment:
Indenting is weird here, can you run the formatter?

Reply:
Done